### DISCUSSION TITLE
What do you want to see?

### DISCUSSION THREAD

**Discussion by johnmarktaylor91** [johnmarktaylor91/torchlens#2] [Ideas]
Thank you for checking out torchlens! As it is still in its early days, I would love to get people's feedback for what they would find useful from it. Generally speaking, the aim of torchlens is to easily extract activations and understand a model's computational graph via visualizations and helpful metadata about a network and its layers. I am unlikely to add functionality for further analyzing the extracted activations, since there already exist excellent packages (e.g., [rsatoolbox](https://rsatoolbox.readthedocs.io/en/latest/index.html)) for this, or for loading particular models (since torchlens is meant to work for any model), but beyond that the door is open for what folks would find useful from it.

**Comment by eric-ayllon** [comment]
Hi,

I don't know where could I mention this so I will leave a comment here. The backward pass visualization (the blue arrows at least) is not properly working in version 0.1.35. I tried visualizing my model, which is a bit large compared to the example in the colab tutorial, and no change in the graph was made after calling backward on the loss. I made sure the gradients were calculated for the model parameters but the gradient visualization was still not working.

Also, adding a couple of the number of gradient tensors (and total size from gradients) to the total stats from the graph could be quite helpful as well.

In the colab you asked whether we see the backpropagation feature as useful, that's why I wanted to mention it.

**Comment by johnmarktaylor91** [comment]
Thanks for bringing this up—do you have the model code handy by any chance, would help me with debugging.

**Reply by eric-ayllon** [reply]
I have the code in a private repository, I could give you access temporarily and provide the execution command I used. I would facilitate the debugging by preparing a branch for you to use if you need it since I am somewhat actively working on that repository. I will send you an email with some details if you are fine with that.

**Comment by johnmarktaylor91** [comment]
Sure, whatever works best—my email is johnmarkedewardtaylor@gmail.com. Just would need the model code to be able to debug
